pkg/vcs/linux_configs.go

```diff
@@ -83,3 +89,60 @@ func linuxConfigsForTags(cf *kconfig.ConfigFile, tags map[string]bool) {
 		}
 	}
 }
+
+// linuxConfigsForType removes Linux kernel config options that are not necessary for the specific
+// report type.
+func linuxConfigsForType(cf *kconfig.ConfigFile, typ dashapi.CrashType, dt debugtracer.DebugTracer) {
+	if typ == dashapi.Warning {
+		typ = dashapi.Bug
+	}
+	keep := map[dashapi.CrashType]func(){
+		dashapi.Hang: func() {
+			cf.Unset("RCU_STALL_COMMON")
+			cf.Unset("LOCKUP_DETECTOR")
+			cf.Unset("DETECT_HUNG_TASK")
+			val := cf.Value("CMDLINE")
+			if val == "" {
+				val = `""`
+			}
+			pos := strings.LastIndexByte(val, '"')
+			if pos >= 0 {
+				cf.Set("CMDLINE",
+					val[:pos]+" rcupdate.rcu_cpu_stall_suppress=1"+val[pos:])
+			}
+		},
+		dashapi.MemoryLeak: func() {
+			cf.Unset("DEBUG_KMEMLEAK")
+		},
+		dashapi.UBSAN: func() {
+			cf.Unset("UBSAN")
+		},
+		dashapi.Bug: func() {
+			cf.Unset("BUG")
+		},
+		dashapi.KASAN: func() {
+			cf.Unset("KASAN")
+		},
+		dashapi.LockdepBug: func() {
+			cf.Unset("LOCKDEP")
+		},
+		dashapi.AtomicSleep: func() {
```

Comment:
Other good bug detectors to disable:
```
KMSAN
KCSAN
DEBUG_OBJECTS
DEBUG_KOBJECT
FORTIFY_SOURCE
REFCOUNT_FULL
SOFTLOCKUP_DETECTOR
```

Maybe also some of:
```
DEBUG_VM*
DEBUG_LIST*
```

At this point I am thinking maybe this should be a part of generic config minimization? If config minimization would be complete, then it would remove all of these, right, and we won't need any of this special code for sanitizers.

Maybe it's possible to just have a list of "more important to remove" configs for minimization, and then the minimization logic will try to remove them more firmly?

Reply:
Also should we remove KASAN/KCSAN/KMSAN/KMEMLEAK/LOCKDEP/etc from baseline configs?
Then even if we have unknown type, or a bug in type detection, we still can remove some during generic minimization.

Reply:
I'd look at it from this perspective: once we are ready to dedicate significant syzbot time doing config minimization, it sounds totally reasonable to drop unneeded sanitizers just like any other on-top-of-baseline options. It looks like we'd have to special case at least the rcu stall/task hung detection case though (it's not a yes/no option), but that should be manageable.

I don't know when exactly we'd find time to restructure syzbot to follow that approach. Unlikely to be soon. So for now we can only do config minimization as a part of fix/cause bisection and therefore we do want to get the most out of the little number of kernel build/boot/test iterations we could take from the actual bug bisection.

And what is really good with sanitizer minimization is that we can infer kernel config options from the bug report quite reliably, so we don't actually need to run those time demanding iterations at all. Well, as it's done in this PR, we can do just one -- to account for the cases when a sanitizer does not catch the bug, but rather makes the kernel slow/big enough for the bug to get triggered. AFAIK that's not a super frequent case though.

Reply:
OK, then add a comment with this and why we are not doing it now.